### PR TITLE
Small improvements and fixes I found a need for

### PR DESCRIPTION
**Pull request by Dead2**
-OpenCL and CUDA is generating hidden invalids, add warnings to make them visible. (Especially noticeable with newer AMD drivers where most results are invalids)
-Cryptonight_lite is missing a test
-Remove a small bit of unused code in amd_gpu/gpu.cpp

I also think the following lines in amd/minethd.cpp and nvidia/minethd.cpp can be removed, but I have no easy way of compile-testing cuda and opencl for xmr-stak, so someone should verify this first.
#include "xmrstak/backend/cpu/crypto/cryptonight_aesni.h"
#include "xmrstak/backend/cpu/crypto/cryptonight.h"

### CODE REVIEW

**Comment by Spudz76** [comment]
Those includes are so that the GPU backends can revalidate incoming "good" results using the CPU miner code to one-time check the result for validity before relaying up to pool.

**Comment by Dead2** [comment]
As I said, I am not entirely certain those can be removed in xmr-stak. But CPU verification works just fine in my fork without them.

File: xmrstak/backend/amd/minethd.cpp [hunk 1]
```diff
@@ -260,9 +260,14 @@ void minethd::work_main()
 
 				hash_fun(bWorkBlob, oWork.iWorkSize, bResult, &cpu_ctx);
 				if ( (*((uint64_t*)(bResult + 24))) < oWork.iTarget)
+				{
 					executor::inst()->push_event(ex_event(job_result(oWork.sJobID, results[i], bResult, iThreadNo, miner_algo), oWork.iPoolId));
+				}
 				else
+				{
+					printer::inst()->print_msg(L3, "AMD Result Invalid");
```
Comment (psychocrypt) [inline xmrstak/backend/amd/minethd.cpp]: Please use `L0` and 
```
const std::string backendName = xmrstak::params::inst().openCLVendor;
printer::inst()->print_msg(L3, "%s GPU (OpenCL) ID = %u Invalid Result", backendName.c_str(), pGpuCtx->deviceIdx);
```

File: xmrstak/backend/nvidia/minethd.cpp [hunk 2]
```diff
@@ -309,9 +309,14 @@ void minethd::work_main()
 
 				hash_fun(bWorkBlob, oWork.iWorkSize, bResult, &cpu_ctx);
 				if ( (*((uint64_t*)(bResult + 24))) < oWork.iTarget)
+				{
 					executor::inst()->push_event(ex_event(job_result(oWork.sJobID, foundNonce[i], bResult, iThreadNo, miner_algo), oWork.iPoolId));
+				}
 				else
+				{
+					printer::inst()->print_msg(L3, "NVIDIA Result Invalid");
```
Comment (psychocrypt) [inline xmrstak/backend/nvidia/minethd.cpp]: Please set to `L0`

File: xmrstak/backend/cpu/minethd.cpp [hunk 3]
```diff
@@ -296,6 +296,7 @@ bool minethd::self_test()
 			bResult = bResult &&  memcmp(out, "\x5a\x24\xa0\x29\xde\x1c\x39\x3f\x3d\x52\x7a\x2f\x9b\x39\xdc\x3d\xb3\xbc\x87\x11\x8b\x84\x52\x9b\x9f\x0\x88\x49\x25\x4b\x5\xce", 32) == 0;
 
 			hashf = func_selector(::jconf::inst()->HaveHardwareAes(), true, xmrstak_algo::cryptonight_lite);
+			hashf("This is a test This is a test This is a test", 44, out, ctx);
```
Comment (psychocrypt) [inline xmrstak/backend/cpu/minethd.cpp]: Could you provide this please a separate PR

File: xmrstak/backend/amd/amd_gpu/gpu.cpp [hunk 4]
```diff
@@ -928,13 +928,6 @@ size_t InitOpenCL(GpuContext* ctx, size_t num_gpus, size_t platform_idx)
 	// create a directory  for the OpenCL compile cache
 	create_directory(get_home() + "/.openclcache");
 
-	// check if cryptonight_monero_v8 is selected for the user or dev pool
-	bool useCryptonight_v8 =
```
Comment (psychocrypt) [inline xmrstak/backend/amd/amd_gpu/gpu.cpp]: please provide this as separate PR

**Comment by Dead2** [comment]
That is an unusual way to handle git PRs, since you can reference individual commits and also revert them separately if needed.  But I'll split it all up into branches and make 3 PRs.

**Pull request closed without merging**